docroot/themes/contrib/civictheme/civictheme_library/components/01-atoms/tag/tag.twig

```diff
@@ -33,25 +34,13 @@
 {% endset %}
 
 {% set tag_text %}
-  {% if icon %}
-    {% if icon_placement == 'before' %}
-      {{ tag_icon }}{{ text }}
-    {% elseif icon_placement == 'after' %}
-      {{ text }}{{ tag_icon }}
-    {% endif %}
-  {% else %}
-    {{ text }}
-  {% endif %}
+  {{ icon_placement == 'before' ? tag_icon : '' }}{{ content }}{{ icon_placement == 'after' ? tag_icon : '' }}
 {% endset %}
 
-{% if tag_text %}
-  <div class="civictheme-tag {{ modifier_class }}" aria-label="{{ text }}">
-    {% if url is not empty %}
-      <a class="civictheme-tag__link" href="{{ url }}" {% if text is not empty %}title="{{ text }}" aria-label="{{ text }}"{% endif%} {{ is_new_window ? 'target="_blank"' }}>
-    {% endif %}
-        {{ tag_text }}
-    {% if url is not empty %}
-      </a>
-    {% endif %}
-  </div>
-{% endif %}
+<div class="civictheme-tag {{ modifier_class }}" aria-label="{{ content }}">
```

Comment:
@govindmaloo 
Could you please:
1. Reinstate the logic that was checking if the content was empty to make sure that no markup is produced if `tag_text` is empty
2. Refactor markup to use `<span>` if not a link or `<a>` (without `<span>` if it is a link)

Reply:
@AlexSkrypnyk  Updated as per the suggestions.